autotest/cpp/test_ogr.cpp

```diff
@@ -2213,4 +2215,108 @@ TEST_F(test_ogr, feature_defn_geomfields_iterator)
     EXPECT_EQ(i, oFDefn.GetGeomFieldCount());
 }
 
+// Test GDALDataset QueryLoggerFunc callback
+TEST_F(test_ogr, GDALDatasetSetQueryLoggerFunc)
+{
+    if (GDALGetDriverByName("GPKG") == nullptr)
+    {
+        GTEST_SKIP() << "GPKG driver missing";
+    }
+
+    auto tmpGPKG{testing::TempDir() + "/poly-1-feature.gpkg"};
```

Comment:
this test should also be skipped if the sqlite version if SQLITE_VERSION_NUMBER < 3014000  . You'll need to include sqlite3.h and add something like the following in autotest/cpp/CMakeLists.txt:
```
target_include_directories(gdal_unit_test PRIVATE $<TARGET_PROPERTY:SQLite::SQLite3,INTERFACE_INCLUDE_DIRECTORIES>)
```